07_errors/jimbotech/storer.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"errors"
+	"strconv"
+)
+
+// Storer defines standard CRUD operations for Puppy
+type Storer interface {
+	CreatePuppy(p *Puppy) (int32, error)
+	ReadPuppy(ID int32) (*Puppy, error)
+	UpdatePuppy(ID int32, Puppy *Puppy) error
+	DeletePuppy(ID int32) error
+}
+
+// Puppy stores puppy details.
+type Puppy struct {
+	ID     int32
+	Breed  string
+	Colour string
+	Value  string
+}
+
+// mapTest used during testing to verify underlaying map changes
+type mapTest interface {
+	length() int
+}
+
+// Error is a custom error
+type Error struct {
+	Message string
+	Code    int
+}
+
+func (e *Error) Error() string {
+	if e == nil {
+		return "<nil>"
+	}
+	return e.Message + " Error Code: " + strconv.Itoa(e.Code)
+}
+
+const (
+	// ErrNegativeID error number if id is negative
+	ErrNegativeID = -1
```

Comment:
Don't set the error code to -1. Just use `iota` and leave the assignment off subsequent error codes. That let's Go assign the numbers since the actual numbers themselves are not relevant.

Reply:
Interesting ... Done